registry/config.go

```diff
@@ -84,16 +84,48 @@ func newServiceConfig(options ServiceOptions) *serviceConfig {
 			IndexConfigs:          make(map[string]*registrytypes.IndexInfo, 0),
 			// Hack: Bypass setting the mirrors to IndexConfigs since they are going away
 			// and Mirrors are only for the official registry anyways.
-			Mirrors: options.Mirrors,
+			Mirrors: make([]string, 0),
 		},
 		V2Only: options.V2Only,
 	}
 
+	config.LoadMirrors(options.Mirrors)
 	config.LoadInsecureRegistries(options.InsecureRegistries)
 
 	return config
 }
 
+// LoadMirrors loads mirrors to config, after removing duplicates.
+// Returns an error if mirrors contains an invalid mirror.
+func (config *serviceConfig) LoadMirrors(mirrors []string) error {
+	mirrorsMap := map[string]struct{}{}
+	uniqueMirrors := []string{}
```

Comment:
note; [variable names in Go](https://github.com/golang/go/wiki/CodeReviewComments#variable-names)